modules/jooby-jakarta-validation/src/main/java/io/jooby/validation/ConstraintViolationHandler.java

```diff
@@ -0,0 +1,93 @@
+package io.jooby.validation;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.jooby.Context;
+import io.jooby.ErrorHandler;
+import io.jooby.StatusCode;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.ConstraintViolationException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.stream.Collectors.groupingBy;
+
+/**
+ * Catches and transform {@link ConstraintViolationException} into {@link ValidationResult}
+ *
+ * Payload example:
+ * <code>
+ * {
+ *     "title": "Validation failed",
+ *     "status": 422,
+ *     "errors": {
+ *         "objectErrors": [
```

Comment:
In gral, I do prefer a single object type where "field" will be nullable/missing. Like:

```json
"errors": [{
   "field": "name" or missing/null,
   "messages": []
   "type": "field" or "global" or "object"
}]
```

Reply:
I tried to make the structure more UI-friendly. UI usually needs field/global errors in separate "baskets". On the other hand, the client is not necessarily UI, so, fair enough.

Let's clarify the types:
- field - it is clear
- global - I suppose it is for cross-field error on the top-level bean
- object - is it for nested beans? in this case, we might need to introduce one more property, cause right now the path for the nested bean errors looks like a full path to the field e.g. `person.firstName`. And in case of the list - `person[0].firstName`

Reply:
It's the first time I see the objectErrors (didn't know about them until now). So I tried to create a fixed structure for clients (UI or not). Clients will mostly want to display the error message and probably the field name. Type is for those clients that want to know more about the error so based on your example type should be `field` (for fieldErrors) or `object` (for objectErrors). Does it make sense now?

Additionally we could render the `path` too which might or might not match `field` attribute